File: src/main/scala/g0001_0100/s0007_reverse_integer/Solution.scala
package g0001_0100.s0007_reverse_integer

// #Medium #Top_Interview_Questions #Math #Udemy_Integers
// #2023_11_09_Time_405_ms_(90.16%)_Space_52.3_MB_(31.15%)

object Solution {
    def reverse(x: Int): Int = {
        var rev: Long = 0
        var remaining = x
        while (remaining != 0) {
            rev = (rev * 10) + (remaining % 10)
            remaining /= 10
        }
        if (rev > Integer.MAX_VALUE || rev < Integer.MIN_VALUE) {
            return 0
        }
        rev.toInt
    }
}

File: src/main/go/g0001_0100/s0007_reverse_integer/solution.go
package s0007_reverse_integer

// #Medium #Top_Interview_Questions #Math #Udemy_Integers
// #2025_04_27_Time_0_ms_(100.00%)_Space_4.03_MB_(24.37%)

import "math"

func reverse(x int) int {
	var r int
	for ok := true; ok; ok = x != 0 {
		if r > math.MaxInt32/10 || r < math.MinInt32/10 {
			return 0
		}
		r = (x % 10) + (r * 10)
		x /= 10
	}
	return r
}
